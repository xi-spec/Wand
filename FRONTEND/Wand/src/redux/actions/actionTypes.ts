export default {
  LOAD_PRODUCT_TYPE: 'LOAD_PRODUCT_TYPE',
  LOAD_ALL_PRODUCTS: 'LOAD_ALL_PRODUCTS',
  LOAD_NEW_PRODUCTS: 'LOAD_NEW_PRODUCTS',
  LOAD_PRODUCTS_LIST: 'LOAD_PRODUCTS_LIST',
  LOAD_PRODUCT_DETAIL: 'LOAD_PRODUCT_DETAIL',
  ADD_PRODUCT_TO_CART: ' ADD_PRODUCT_TO_CART',
  INCREASE_QUANTITY_CART: 'INCREASE_QUANTITY_CART',
  DECREASE_QUANTITY_CART: 'DECREASE_QUANTITY_CART',
  DELETE_PRODUCT_CART: 'DELETE_PRODUCT_CART',
  DELETE_ALL_CART: 'DELETE_ALL_CART',
  UPDATE_STOCK_API: 'UPDATE_STOCK_API',
  SEARCH_PRODUCT: 'SEARCH_PRODUCT',
  ADD_PRODUCT_TO_FAVOURITE: 'ADD_PRODUCT_TO_FAVOURITE',
  REGISTER: 'REGISTER',
  REGISTER_ERROR: 'REGISTER_ERROR',
  LOGIN: 'LOGIN',
  LOGIN_ERROR: 'LOGIN_ERROR',
  CLEAN_ISEXIST_STATUS: 'CLEAN_STATUS',
  CLEAN_LOGIN_ERROR: 'CLEAN_LOGIN_ERROR',
  CLEAN_REGISTED: 'CLEAN_REGISTED',
  REMOVE_FAVOURITE: 'REMOVE_FAVOURITE',
  LOGOUT: 'LOGOUT',
  UPDATE_USER: 'UPDATE_USER',
  PAYMENT_SUCCSSFULY: 'PAYMENT_SUCCESSFULY',
  PDATE_STOCK_API: 'PDATE_STOCK_API'
}
;
