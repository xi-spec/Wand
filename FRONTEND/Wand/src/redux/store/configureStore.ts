import reduxImmutableStatInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import { applyMiddleware, compose, createStore } from 'redux';
import rootReducer from '../reducers';

function configureStore () {
  const composeEnhancers = compose;

  return createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(reduxImmutableStatInvariant(), thunk))
  );
}

export default configureStore();
