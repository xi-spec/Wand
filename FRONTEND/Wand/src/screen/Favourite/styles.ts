
import { StyleSheet } from 'react-native';
import { SIZE, FONTS, COLORS } from '../../constants/theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    margin: 10
  },

  shadowBox: {
    height: 200,
    justifyContent: 'center',
    marginTop: 10,
    shadowColor: 'grey',
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.15,
    shadowRadius: 3.84,
    backgroundColor: 'white',
    elevation: 5
  },
  listContainer: {
    flexDirection: 'row'
  },
  textContainer: {

  },
  detail: {
    fontSize: SIZE.primary,
    fontFamily: FONTS.logoRegular,
    marginBottom: 5,
    paddingTop: 5
  },
  motto: {
    fontSize: SIZE.second,
    fontFamily: FONTS.extraLight,
    paddingTop: 5
  },
  rowContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  remove: {
    paddingBottom: 45,
    paddingLeft: 90
  },
  cost: {
    marginLeft: 50,
    paddingBottom: 15,
    fontSize: SIZE.primary,
    color: COLORS.second,
    fontFamily: FONTS.primary
  },
  imageContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    margin: 5,
    width: '60%'
  },
  columText: {
    marginLeft: 10
  },
  image: {
    justifyContent: 'center',
    width: 100,
    height: 150,
    borderRadius: 5

  },

  addContainer: {
    width: 120,
    height: 35,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: COLORS.primary,
    alignContent: 'center',
    borderRadius: 5
  },
  disabled: {
    width: 120,
    height: 35,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: COLORS.lightGrey,
    alignContent: 'center',
    borderRadius: 5
  },
  totalContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginHorizontal: 5
  },
  totalText: {
    fontSize: SIZE.primary,
    fontFamily: FONTS.primary
  },
  addtext: {
    color: 'white',
    fontFamily: FONTS.primary,
    fontSize: SIZE.second

  }

});

export default styles
;
