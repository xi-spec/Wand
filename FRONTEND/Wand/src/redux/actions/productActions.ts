import actionTypes from './actionTypes';
import axios from 'axios';
import SERVER from '../../constants/server';
import { Dispatch } from 'redux';
import { Product, ShoppingCart } from '../../models/initiaState';

export function loadProductsType () {
  return async (dispatch : Dispatch) => {
    const { data } = await axios.get(SERVER.URL_TYPE);
    dispatch({
      type: actionTypes.LOAD_PRODUCT_TYPE,
      data
    });
  };
}

export function loadAllProducts () {
  return async (dispatch : Dispatch) => {
    const { data } = await axios.get(SERVER.URL_PRODUCT);
    dispatch({
      type: actionTypes.LOAD_ALL_PRODUCTS,
      data
    });
  };
}

export function loadProductsList (typeName:string) {
  return {
    type: actionTypes.LOAD_PRODUCTS_LIST,
    typeName
  };
}

export function loadProductDetail (productId:string) {
  return {
    type: actionTypes.LOAD_PRODUCT_DETAIL,
    productId
  };
}

export function addProductToCart (product:Product) {
  return {
    type: actionTypes.ADD_PRODUCT_TO_CART,
    product
  };
}

export function increaseQuantityCart (productId:string) {
  return {
    type: actionTypes.INCREASE_QUANTITY_CART,
    productId
  };
}

export function decreaseQuantityCart (productId:string) {
  return {
    type: actionTypes.DECREASE_QUANTITY_CART,
    productId
  };
}

export function deleteProductCart (productId:string, quantity:number) {
  return {
    type: actionTypes.DELETE_PRODUCT_CART,
    productId,
    quantity
  };
}

export function searchProducts (name:string) {
  return {
    type: actionTypes.SEARCH_PRODUCT,
    name
  };
}

export function updateStockApi (cart:ShoppingCart) {
  return (dispatch : Dispatch) => {
    cart.cartList.forEach(async (product) => {
      const { data } = await axios.put(`${SERVER.URL_PRODUCT}/${product._id}`, { stock: product.stock });
      dispatch({
        type: actionTypes.UPDATE_STOCK_API,
        data
      });
    });
  };
}
