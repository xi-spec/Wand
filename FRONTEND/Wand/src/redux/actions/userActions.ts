import axios from 'axios';
import actionTypes from './actionTypes';
import SERVER from '../../constants/server';
import { User, Product, ShoppingCart } from '../../models/initiaState';
import { Dispatch } from 'redux';

export function addProductToFavourite (product:Product, user:User, favouriteState:boolean) {
  let newUser;
  favouriteState
    ? newUser = { ...user, favourite: [...user.favourite, product] }
    : newUser = { ...user, favourite: user.favourite.filter(elem => elem._id !== product._id) };

  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, newUser);
    dispatch({
      type: actionTypes.ADD_PRODUCT_TO_FAVOURITE,
      data
    });
  };
}

export function register (email:string, name:string, password:string) {
  const query = { email, name, password };
  let actionObj;
  return async (dispatch:Dispatch) => {
    try {
      const { data, status } = await axios.post(SERVER.URL_REGISTER, query);
      actionObj = {
        type: actionTypes.REGISTER,
        data,
        status
      };
    } catch (error) {
      actionObj = {
        type: actionTypes.REGISTER_ERROR
      };
    }
    dispatch(
      actionObj
    );
  };
}

export function login (email:string, password:string) {
  const query = { email, password };

  return async (dispatch:Dispatch) => {
    let actionObj;
    try {
      const { data } = await axios.post(SERVER.URL_LOGIN, query);
      actionObj = {
        type: actionTypes.LOGIN,
        data

      };
    } catch (error) {
      actionObj = {
        type: actionTypes.LOGIN_ERROR
      };
    }

    dispatch(actionObj);
  };
}

export function logout (email:string, password:string) {
  const query = { email, password };

  return async (dispatch:Dispatch) => {
    const { data } = await axios.post(SERVER.URL_LOGOUT, query);

    dispatch({
      type: actionTypes.LOGOUT,
      data
    });
  };
}

export function cleanIsExistStatus () {
  return {
    type: actionTypes.CLEAN_ISEXIST_STATUS
  };
}

export function cleanLoginError () {
  return {
    type: actionTypes.CLEAN_LOGIN_ERROR
  };
}

export function cleanRegisted () {
  return {
    type: actionTypes.CLEAN_REGISTED
  };
}

export function removeFavourite (product:Product, user:User) {
  const newUser = { ...user, favourite: user.favourite.filter(elem => elem._id !== product._id) };

  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, newUser);
    dispatch({
      type: actionTypes.REMOVE_FAVOURITE,
      data
    });
  };
}

export function updateAddress (street:string,
  city:string, country:string, postcode:string, user:User) {
  const query = { street, city, country, postcode };
  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, { address: query });
    dispatch({
      type: actionTypes.UPDATE_USER,
      data
    });
  };
}

export function updateCard (cardNumber:string, securtyCode:string, cardholder:string, expiryDate:string, user:User) {
  const query = { cardNumber, securtyCode, cardholder, expiryDate };
  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, { card: query });
    dispatch({
      type: actionTypes.UPDATE_USER,
      data
    });
  };
}

export function paymentSuccessfuly (user:User, cart:ShoppingCart) {
  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, { orderHistory: [...user.orderHistory, cart] });
    dispatch({
      type: actionTypes.PAYMENT_SUCCSSFULY,
      data
    });
  };
}
