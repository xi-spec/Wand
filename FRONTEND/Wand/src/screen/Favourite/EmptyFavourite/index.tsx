import React from 'react';
import { View, Text, Image, TouchableOpacity } from 'react-native';
import styles from './styles';
import { IMAGES } from '../../../constants/images';
import { FAVOURITE_TEXT } from '../../../constants/screenTexts';

export default function EmptyFavourite ({ navigation }:{navigation:
  {navigate:Function, reset:Function}}) {
  return (
        <View style={styles.container}>
          <Image style={styles.image}source={{ uri: IMAGES.emptyFavourite }}></Image>
        <Text style={styles.text}>{FAVOURITE_TEXT.EMPTY_FAVOURITE_TITLE}</Text>
        <Text style={styles.text}>{FAVOURITE_TEXT.EMPTY_FAVOURITE_SUBTITLE}</Text>

        <View style={styles.buttonsContainer}>

        <TouchableOpacity
        testID='shopNow'
        style={{ ...styles.button, ...styles.colorButton }}
        onPress={() => navigation.reset({
          index: 0,
          routes: [{ name: 'Home' }]
        })}>
            <Text style={styles.buttonText}>SHOP NOW</Text>
          </TouchableOpacity>

          <TouchableOpacity
          testID="viewCart"
           style={{ ...styles.button, ...styles.whiteButton }}
           onPress={() => navigation.navigate('Cart')}>
            <Text style={{ ...styles.buttonText, ...styles.blackText }}>VIEW CART</Text>
          </TouchableOpacity>
        </View>

      </View>

  );
}
