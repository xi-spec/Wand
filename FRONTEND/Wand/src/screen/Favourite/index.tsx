import React from 'react';
import {
  Text,
  View,
  FlatList,
  TouchableOpacity,
  Image
} from 'react-native';
import EmptyFavourite from './EmptyFavourite';
import styles from './styles';
import { Icon } from 'react-native-elements';
import { connect } from 'react-redux';
import { User } from '../../models/initiaState';
import { bindActionCreators } from 'redux';
import { addProductToCart } from '../../redux/actions/productActions';
import { removeFavourite } from '../../redux/actions/userActions';
import setCostValue2Decimals from '../../utils/setCostValue2Decimals';

function Favourite ({ navigation, user, actions }:{navigation:{navigate:Function, reset:Function}, user:User, actions:any}) {
  const totalCost = () => {
    return user.favourite.reduce((acc, product) => acc + product.cost.value, 0).toFixed(2);
  };
  return (
    user.favourite.length === 0
      ? <EmptyFavourite navigation={navigation}/>
      : <View style={styles.container}>

      <FlatList
      bounces={false}
      data={user.favourite}
      numColumns={1}
      keyExtractor={(item) => item._id.toString()}
      showsVerticalScrollIndicator={false}
      renderItem={({ item }) =>
      <View style={styles.shadowBox}>
      <View style={styles.listContainer}>
        <TouchableOpacity
        style={styles.imageContainer}
        key={Math.random() * 99}
        testID='favourite'
        onPress={() => navigation.navigate('Detail', { productId: item._id })}>
      <Image source={{ uri: item.images.alls[0] }}
      style={styles.image} />
          <View style={styles.columText}>
        <Text style={styles.detail}>{item.name}</Text>
        <Text style={styles.motto}> What a Nice Day!</Text>
          </View>
        </TouchableOpacity>

        <View style={styles.textContainer}>
        <TouchableOpacity
        testID='remove'
        style={styles.remove}
        onPress={() => actions.removeFavourite(item, user)}>
               <Icon name='close-outline' type="ionicon"></Icon>
             </TouchableOpacity>
             <Text style={styles.cost}>{`${setCostValue2Decimals(item.cost.value)} ${item.cost.currency}`}</Text>
        <TouchableOpacity
        testID='add'
        disabled={!item.stock}
        onPress={() => { actions.addProductToCart(item); actions.removeFavourite(item, user); }}
        style={ item.stock ? styles.addContainer : styles.disabled}>
              <Text style={styles.addtext}>ADD TO CART</Text>
            </TouchableOpacity>
        </View>
        </View>
      </View>
        }>
   </FlatList>
   <View style={styles.totalContainer}>
           <Text style={styles.totalText}>{`${user.favourite.length} items`}</Text>
           <Text style={styles.totalText}>{`${totalCost()} euros`}</Text>
   </View>
   </ View >

  );
}

function mapStateToProps ({ user }) {
  return {
    user
  };
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({
      addProductToCart,
      removeFavourite
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Favourite);
