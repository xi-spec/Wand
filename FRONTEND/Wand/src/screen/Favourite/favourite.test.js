import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import configureStore from 'redux-mock-store';
import * as actions from '../../redux/actions/userActions';
import * as productActions from '../../redux/actions/productActions';
import { Provider } from 'react-redux';
import Favourite from './index';

describe('Given a Favourite component', () => {
  const mockStore = configureStore();
  let store;
  let navigation;

  beforeEach(() => {
    jest.spyOn(actions, 'removeFavourite').mockReturnValueOnce({ type: '' });
    jest.spyOn(productActions, 'addProductToCart').mockReturnValueOnce({ type: '' });

    store = mockStore({
      user: {
        favourite: [
          {
            name: 'ring',
            cost: {
              value: 1,
              currency: 'euro'
            },
            _id: 1,
            images: {
              alls: ['firstImageUrl']
            }
          }]
      }
    });
    navigation = {
      navigate: jest.fn()
    };
  });
  describe('When is invoked', () => {
    test('Then will rendered correclly', () => {
      const tree = render(<Provider store={store}><Favourite /></Provider>);

      expect(tree).toMatchSnapshot();
    });
  });
  describe('When is invoked with empty user.favourite ', () => {
    test('Then will rendered correclly', () => {
      store = mockStore({
        user: {
          favourite: []
        }
      });
      const tree = render(<Provider store={store}><Favourite /></Provider>);

      expect(tree).toMatchSnapshot();
    });
  });
  describe('When is pressed in favourite button', () => {
    test('Then navigation.navigate is invoked', () => {
      const { getByTestId } = render(<Provider store={store}><Favourite navigation={navigation} /></Provider>);

      fireEvent.press(getByTestId('favourite'));

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });

  describe('When is pressed in add button', () => {
    test('Then productActions.addProductToCart is invoked', () => {
      const { getByTestId } = render(<Provider store={store}><Favourite navigation={navigation} /></Provider>);

      fireEvent.press(getByTestId('add'));

      expect(productActions.addProductToCart).toHaveBeenCalled();
    });
  });

  describe('When is pressed in remove button', () => {
    test('Then actions.removeFavourite is invoked', () => {
      const { getByTestId } = render(<Provider store={store}><Favourite navigation={navigation} /></Provider>);

      fireEvent.press(getByTestId('remove'));

      expect(actions.removeFavourite).toHaveBeenCalled();
    });
  });
});
