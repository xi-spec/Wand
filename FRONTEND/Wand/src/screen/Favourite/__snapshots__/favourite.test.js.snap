// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Favourite component When is invoked Then will rendered correclly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "margin": 10,
    }
  }
>
  <RCTScrollView
    bounces={false}
    data={
      Array [
        Object {
          "_id": 1,
          "cost": Object {
            "currency": "euro",
            "value": 1,
          },
          "images": Object {
            "alls": Array [
              "firstImageUrl",
            ],
          },
          "name": "ring",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "elevation": 5,
              "height": 200,
              "justifyContent": "center",
              "marginTop": 10,
              "shadowColor": "grey",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.15,
              "shadowRadius": 3.84,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "width": "45%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "margin": 5,
                  "opacity": 1,
                }
              }
              testID="favourite"
            >
              <Image
                source={
                  Object {
                    "uri": "firstImageUrl",
                  }
                }
                style={
                  Object {
                    "borderRadius": 5,
                    "height": 150,
                    "justifyContent": "center",
                    "width": 100,
                  }
                }
              />
              <View
                style={
                  Object {
                    "marginLeft": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontFamily": "RalewayRegular",
                      "fontSize": 18,
                      "marginBottom": 5,
                      "paddingTop": 5,
                    }
                  }
                >
                  ring
                </Text>
                <Text
                  style={
                    Object {
                      "fontFamily": "RalewayExtraLight",
                      "fontSize": 14,
                      "paddingTop": 5,
                    }
                  }
                >
                   What a Nice Day!
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "paddingLeft": 20,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                    "paddingBottom": 45,
                    "paddingLeft": 90,
                  }
                }
                testID="remove"
              >
                <View
                  style={
                    Object {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </View>
              </View>
              <Text
                style={
                  Object {
                    "color": "#F3603F",
                    "fontFamily": "MontserratRegular",
                    "fontSize": 18,
                    "marginLeft": 50,
                    "paddingBottom": 15,
                  }
                }
              >
                1.00 euro
              </Text>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignContent": "center",
                    "alignItems": "center",
                    "backgroundColor": "#2E3192",
                    "borderRadius": 5,
                    "height": 35,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": 120,
                  }
                }
                testID="add"
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "MontserratRegular",
                      "fontSize": 14,
                    }
                  }
                >
                  ADD TO CART
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "MontserratRegular",
          "fontSize": 18,
        }
      }
    >
      1 items
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "MontserratRegular",
          "fontSize": 18,
        }
      }
    >
      1.00 euros
    </Text>
  </View>
</View>
`;

exports[`Given a Favourite component When is invoked with empty user.favourite  Then will rendered correclly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingTop": 266.8,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://trello-attachments.s3.amazonaws.com/603be1b68ce9c654d30ae543/60506d7b3b22568450a0dc85/9d436ab3753190846db49e3ce1cf6950/final1-1024x1024.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 50,
        "height": 100,
        "width": 100,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "RalewayRegular",
        "fontSize": 14,
        "marginHorizontal": 25,
        "paddingVertical": 20,
      }
    }
  >
    YOUR WISHLIST IS EMPTY
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "RalewayRegular",
        "fontSize": 14,
        "marginHorizontal": 25,
        "paddingVertical": 20,
      }
    }
  >
    Check if there any products in your basket and snatch them up before they're gone
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "bottom": 266.8,
        "height": 80,
        "position": "absolute",
        "width": "90%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2E3192",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
      testID="shopNow"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        SHOP NOW
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
      testID="viewCart"
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        VIEW CART
      </Text>
    </View>
  </View>
</View>
`;
